dynamic_programming/regular_expression_matching.py

```diff
@@ -0,0 +1,39 @@
+"""
+Given an input string (s) and a pattern (p), implement regular expression matching with support for '.' and '*' where: 
+
+'.' Matches any single character.
+'*' Matches zero or more of the preceding element.
+The matching should cover the entire input string (not partial).
+
+https://leetcode.com/problems/regular-expression-matching/
+"""
+def isMatch(self, s: str, p: str) -> bool:
```

Comment:
Python naming conventions state that function names should be in snake case and one letter variable names are not acceptable. Also, there is no need for a `self` argument as the function isn't a class method:

```python
def is_match(check_string: str, pattern: str) -> bool:
    ...
```